fetch.bs

```diff
@@ -2204,6 +2204,47 @@ unset or <a for=request>keepalive</a> is false, <a lt=terminated for=fetch>termi
 <a for="fetch group">fetch record</a>'s <a for="fetch record">fetch</a>.
 
 
+<h3 id=resolving-domains>Resolving domains</h3>
+
+<p tracking-vector>To <dfn>resolve a domain</dfn>, given a <a for=/>domain</a> <var>domain</var> and
+a <a for=/>network partition key</a> <var>key</var>, perform an <a>implementation-defined</a>
+operation to turn <var>domain</var> into a <a for=/>set</a> of one or more
+<a for=/>IP addresses</a>. If this operation succeeds, return the <a for=/>set</a> of
+<a for=/>IP addresses</a>. If it fails, return failure. The results of this operation may be cached.
+If they are cached, <var>key</var> should be used as part of the cache key.
+
+<div class=note>
+ <p>Typically this operation would involve DNS and as such caching can happen on DNS servers without
+ <var>key</var> being taken into account. Depending on the implementation it might also not be
+ possible to take <var>key</var> into account locally. [[RFC1035]]
+
+ <p>Callers ought to not rely on the order of the <a for=/>set</a> <a>resolve a domain</a> can
```

Comment:
We take it as a hint for what order to try connecting to the IP addresses in. It's okay to use it in that way, even though the order may end up being different from what the server intended.

A couple of traditional ways of setting the order that results are returned in:
* Return the "main" web server as the first item in the list, and the "backup" web server second.
* Round-robin the results to balance load over a cluster of servers.